examples/air-purifier-app/air-purifier-common/src/air-purifier-manager.cpp

```diff
@@ -0,0 +1,312 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <air-purifier-manager.h>
+#include <app/util/error-mapping.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using Protocols::InteractionModel::Status;
+
+void AirPurifierManager::Init()
+{
+    FanControl::SetDefaultDelegate(mEndpointId, this);
+
+    activatedCarbonFilterInstance.Init();
+    hepaFilterInstance.Init();
+    mAirQualitySensorManager.Init();
+    mTemperatureSensorManager.Init();
+    mHumiditySensorManager.Init();
+
+    DataModel::Nullable<Percent> percentSetting;
+    EmberAfStatus status = FanControl::Attributes::PercentSetting::Get(mEndpointId, percentSetting);
+    if (EMBER_ZCL_STATUS_SUCCESS == status)
+    {
+        if (percentSetting.IsNull())
+        {
+            PercentSettingWriteCallback(0);
+        }
+        else
+        {
+            PercentSettingWriteCallback(percentSetting.Value());
+        }
+    }
+
+    DataModel::Nullable<uint8_t> speedSetting;
+    status = FanControl::Attributes::SpeedSetting::Get(mEndpointId, speedSetting);
+    if (EMBER_ZCL_STATUS_SUCCESS == status)
+    {
+        if (speedSetting.IsNull())
+        {
+            SpeedSettingWriteCallback(0);
+        }
+        else
+        {
+            SpeedSettingWriteCallback(speedSetting.Value());
+        }
+    }
+
+    // Set up some sane initial values for temperature and humidity - note these are fixed values for testing purposes only
+    // TODO: Make these values settable via linux pipe
```

Comment:
Is this TODO in the correct place given that this is under the common files (not linux specific)